File: src/main/webapp/scripts/app/game/engine/token/Direction.js
var Direction = {

    NORTH: "NORTH",
    WEST: "WEST",
    SOUTH: "SOUTH",
    EAST: "EAST",

    getDirection: function(directionString){
        switch(directionString){
            case "NORTH": return Direction.NORTH;
            case "WEST": return Direction.WEST;
            case "SOUTH": return Direction.SOUTH;
            case "EAST": return Direction.EAST;
        }
        return Direction.NORTH;
    },
    /**
     * returns 1 for turning left and -1 for turning right
     */
    getKonvaRotation: function(fromDirection, toDirection){

        switch(fromDirection){
            case Direction.NORTH:
                if(toDirection == Direction.WEST){return -180;}
                if(toDirection == Direction.EAST){return 0;}
                if(toDirection == Direction.SOUTH){return 90;}
                break;
            case Direction.WEST:
                if(toDirection == Direction.SOUTH){return -270;}
                if(toDirection == Direction.NORTH){return 270;}
                if(toDirection == Direction.EAST){return 0;}
                break;
            case Direction.SOUTH:
                if(toDirection == Direction.EAST){return -360;}
                if(toDirection == Direction.WEST){return 180;}
                if(toDirection == Direction.NORTH){return 270;}
                break;
            case Direction.EAST:
                if(toDirection == Direction.NORTH){return -90;}
                if(toDirection == Direction.SOUTH){return 90;}
                if(toDirection == Direction.WEST){return 180;}
                break;
        }

        return 1;

    }


}

File: src/main/java/de/overwatch/gungungun/game/util/Direction.java
package de.overwatch.gungungun.game.util;

import de.overwatch.gungungun.game.model.Coordinate;

public enum Direction {
	NORTH, 
	EAST, 
	SOUTH, 
	WEST;

	public static Direction getDirection( String directionString ){
		for( Direction direction: Direction.values() ){
			if( direction.name().equals(  directionString ) ){
				return direction;
			}
		}
		throw new IllegalArgumentException( "There is no Direction defined with name = "+directionString );
	}
	public static final Direction get180DegreeDirection( Direction direction) {
		if( direction.equals( Direction.NORTH ))
			return Direction.SOUTH;
		if( direction.equals( Direction.SOUTH ))
			return Direction.NORTH;
		if( direction.equals( Direction.WEST ))
			return Direction.EAST;
		if( direction.equals( Direction.EAST ))
			return Direction.WEST;
		throw new RuntimeException("not a valid Direction");
	}
	/**	
	 * 	Turns a Direction on a Coordinate towards another Coordinate
	 * 
	 * 	Does nothing, if the Coordinates are the same
	 */
	public static final Direction getDirectionTowardsCoordinate(Coordinate ownCoordinate, Coordinate otherCoordinate){
		int x = ownCoordinate.getX() - otherCoordinate.getX();
		int y = ownCoordinate.getY() - otherCoordinate.getY();
		if(Math.abs(x) < Math.abs(y)){
			if(y>0)
				return Direction.NORTH;
			else if (y<0)
				return Direction.SOUTH;
		} else if(Math.abs(x) > Math.abs(y)){ 
			if(x>0)
				return Direction.WEST;
			else if (x<0)
				return Direction.EAST;
		}else if(Math.abs(x) == Math.abs(y)){
            if( y < 0 ){
                return Direction.SOUTH;
            }
            if( y > 0 ){
                return Direction.NORTH;
            }
        }
		//same Coordinate
		return Direction.NORTH;
	}
	public static final Direction getTurnRightDirection( Direction currentDirection ){
		if( currentDirection.equals( Direction.NORTH ))
			return Direction.EAST;
		if( currentDirection.equals( Direction.SOUTH ))
			return Direction.WEST;
		if( currentDirection.equals( Direction.WEST ))
			return Direction.NORTH;
		if( currentDirection.equals( Direction.EAST ))
			return Direction.SOUTH;
		throw new RuntimeException("not a valid Direction");
	}
	
	public static final Direction getTurnLeftDirection( Direction currentDirection ){
		if( currentDirection.equals( Direction.NORTH ))
			return Direction.WEST;
		if( currentDirection.equals( Direction.SOUTH ))
			return Direction.EAST;
		if( currentDirection.equals( Direction.WEST ))
			return Direction.SOUTH;
		if( currentDirection.equals( Direction.EAST ))
			return Direction.NORTH;
		throw new RuntimeException("not a valid Direction");
	}
}
